wpilibc/shared/include/SmartDashboard/SendableChooser.h

```diff
@@ -27,24 +27,34 @@ namespace frc {
  * laptop.  Once autonomous starts, simply ask the {@link SendableChooser} what
  * the selected value is.</p>
  *
+ * @tparam T The type of values to be stored
  * @see SmartDashboard
  */
+template <class T>
 class SendableChooser : public Sendable {
  public:
   virtual ~SendableChooser() = default;
 
-  void AddObject(const std::string& name, void* object);
-  void AddDefault(const std::string& name, void* object);
-  void* GetSelected();
+  void AddObject(const std::string& name, T& object);
+  void AddObject(const std::string& name, const T& object);
+  void AddDefault(const std::string& name, T& object);
+  void AddDefault(const std::string& name, const T& object);
+  T GetSelected();
 
   void InitTable(std::shared_ptr<ITable> subtable) override;
   std::shared_ptr<ITable> GetTable() const override;
   std::string GetSmartDashboardType() const override;
 
  private:
+  static const std::string kDefault;
+  static const std::string kOptions;
+  static const std::string kSelected;
+
   std::string m_defaultChoice;
-  std::map<std::string, void*> m_choices;
+  std::map<std::string, T> m_choices;
   std::shared_ptr<ITable> m_table;
```

Comment:
We could refactor some of these as well to a non template base class, but that's a nice to have not a strict requirement.